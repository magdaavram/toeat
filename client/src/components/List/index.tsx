import React from 'react';
import styled from 'styled-components';

type Mapper = (item: any) => string;

const ListContainer = styled.ul`
  padding: 0;
  list-style: none;
`;

const Item = styled.li`
  padding: 9px;
  border-bottom: 2px solid rgba(154, 140, 152, 0.2);

  &:before {
    content: '\\25CF';
    color: var(--color--burgundy);
    font-size: var(--font-size--small);
    opacity: 0.8;
    margin-right: 9px;
    padding-bottom: 3px;
    display: inline-block;
    vertical-align: middle;
  }
`;

const List = ({ items, renderItem }: { items: any[]; renderItem: Mapper }) => {
  return (
    <ListContainer>
      {items.map((item: any, index: number) => (
        <Item key={`item-${index}`}>{renderItem(item)}</Item>
      ))}
    </ListContainer>
  );
};

export default List;
