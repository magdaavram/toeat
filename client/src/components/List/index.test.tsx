import React from 'react';
import { mount } from 'enzyme';
import List from './index';

it('should render correctly', () => {
  const props = {
    items: [
      {
        ingredient: 'tomatoes',
        quantity: 200,
        unit: 'grams',
      },
      {
        ingredient: 'potatoes',
        quantity: 500,
        unit: 'grams',
      },
    ],
    renderItem: (ingredient) => `${ingredient.quantity} ${ingredient.unit} of ${ingredient.ingredient}`,
  };

  const wrapper = mount(<List {...props}/>);

  expect(wrapper).toMatchSnapshot();
});