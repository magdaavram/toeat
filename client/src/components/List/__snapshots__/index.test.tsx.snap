// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<List
  items={
    Array [
      Object {
        "ingredient": "tomatoes",
        "quantity": 200,
        "unit": "grams",
      },
      Object {
        "ingredient": "potatoes",
        "quantity": 500,
        "unit": "grams",
      },
    ]
  }
  renderItem={[Function]}
>
  <styled.ul>
    <ul
      className="sc-AxjAm ksBuCE"
    >
      <styled.li
        key="item-0"
      >
        <li
          className="sc-AxirZ frEcip"
        >
          200 grams of tomatoes
        </li>
      </styled.li>
      <styled.li
        key="item-1"
      >
        <li
          className="sc-AxirZ frEcip"
        >
          500 grams of potatoes
        </li>
      </styled.li>
    </ul>
  </styled.ul>
</List>
`;
